### PR TITLE
New scan/audit pretty-json output format

### PR DESCRIPTION
**Pull request by RobiNino**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----
    
The new format transforms the raw scan results into a "table like" json.
```
{
  "Vulnerabilities": [],
  "SecurityViolations": [],
  "LicensesViolations": [],
  "Licenses": []
}
```
Each table is an array of key:value objects, with the keys corresponding to the column names in the `table` format.

### CODE REVIEW

File: utils/coreutils/tableutils.go [hunk 1]
```diff
@@ -88,28 +90,58 @@ var DefaultMaxColWidth = 25
...
+// ┌──────┬─────┐
+// │ Name │ Age │
+// ├──────┼─────┤
+// │ Gai  │ 350 │
+// ├──────┼─────┤
+// │ Noah │ 21  │
+// └──────┴─────┘
+// Becomes:
+// {[{"Name": "Gai", "Age": "350"}, {"Name": "Noah", "Age": "21"}]}
+func CreateJsonTable(rows interface{}) ([]map[string]interface{}, error) {
```
Comment (eyalbe4) [inline utils/coreutils/tableutils.go:123]: The new simple-json format shouldn't be affected by future changes to the table format.
We should therefore probably remove this function (which creates the simple-json output) out of the tableutils.go, which includes the code for creating the tables.

File: utils/coreutils/utils.go [hunk 2]
```diff
@@ -400,3 +400,17 @@ func SetCliExecutableName(executableName string) {
 func GetCliExecutableName() string {
 	return cliExecutableName
 }
+
+// Get the explicit list by number of elements.
+// For example for multiple elements: "one, two and three".
+// For a single element: "one".
+func GetExplicitListByNumber(list []string) string {
```
Comment (eyalbe4) [inline utils/coreutils/utils.go:407]: GetExplicitListByNumber --> ListToText

File: utils/coreutils/utils.go [hunk 3]
```diff
@@ -400,3 +400,17 @@ func SetCliExecutableName(executableName string) {
 func GetCliExecutableName() string {
 	return cliExecutableName
 }
+
+// Get the explicit list by number of elements.
```
Comment (eyalbe4) [inline utils/coreutils/utils.go:404]: Get the explicit list by number of elements.
-->
Turn a list of strings into a sentence.

File: utils/coreutils/utils.go [hunk 4]
```diff
@@ -400,3 +400,17 @@ func SetCliExecutableName(executableName string) {
...
+// For example for multiple elements: "one, two and three".
```
Comment (eyalbe4) [inline utils/coreutils/utils.go:405]: For example for multiple elements: "one, two and three".
-->
For example, turn ["one","two", "three"] into "one, two and three".

File: xray/utils/resultstable.go [hunk 5]
```diff
@@ -202,6 +258,13 @@ type cveRow struct {
 	cvssV3 string `col-name:"CVSS\nv3" extended:"true"`
 }
 
+type ResultsJsonTable struct {
```
Comment (eyalbe4) [inline xray/utils/resultstable.go:261]: ResultsJsonTable --> ResultsSimpleJson

Let's also add a description for this struct, explaining its role.

File: xray/utils/resultwriter.go [hunk 6]
```diff
@@ -8,27 +8,26 @@ import (
...
 
 type OutputFormat string
 
 const (
 	// OutputFormat values
-	Table OutputFormat = "table"
-	Json  OutputFormat = "json"
+	Table  OutputFormat = "table"
+	Json   OutputFormat = "json"
+	Pretty OutputFormat = "pretty-json"
```
Comment (eyalbe4) [inline xray/utils/resultwriter.go:21]: pretty-json --> simple-json

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
The requested changes in this PR can also be implemented in a follow-up PR.

File: utils/coreutils/tableutils.go (hunk 1 above)
Comment (RobiNino) [reply utils/coreutils/tableutils.go:123]: Will be fixed in a follow-up PR

**Pull request merged**